### PR TITLE
Put type_of_array in inspector

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

File: include/highfive/bits/H5Utils.hpp [hunk 1]
```diff
@@ -37,10 +37,15 @@ template <std::size_t N>
...
 
+template <typename T>
+using unqualified_t = typename std::remove_const<typename std::remove_reference<T>::type
+        >::type;
+
 namespace details {
 template <typename T>
 struct manipulator {
     using type = T;
+    using hdf5_type = unqualified_t<T>;
```
Comment (ferdonline) [inline include/highfive/bits/H5Utils.hpp:48]: to me this is a `base_type` from a composite type. Is there any connection to hdf5?

Comment (alkino) [reply include/highfive/bits/H5Utils.hpp:48]: hdf5_type is the type than HDF5 want to see internally.
the type behind the `void*` in `write(void* value)`

Comment (alkino) [reply include/highfive/bits/H5Utils.hpp:48]: I don't remember right now when, but sometimes it differs from the `h5_type` (futur) that is the type to store the array.

**Review by ohm314 — APPROVED** [review]

File: include/highfive/bits/H5Utils.hpp (hunk 1 above)
Comment (alkino) [reply include/highfive/bits/H5Utils.hpp:48]: but base_type can be a good name.

File: include/highfive/bits/H5Utils.hpp [hunk 2]
```diff
@@ -37,10 +37,15 @@ template <std::size_t N>
 class FixedLenStringArray;
 
 
+template <typename T>
+using unqualified_t = typename std::remove_const<typename std::remove_reference<T>::type
+        >::type;
```
Comment (matz-e) [inline include/highfive/bits/H5Utils.hpp:42]: That looks…weird

Comment (ohm314) [reply include/highfive/bits/H5Utils.hpp:42]: looks fine to me?

Comment (ohm314) [reply include/highfive/bits/H5Utils.hpp:42]: with C++20: https://en.cppreference.com/w/cpp/types/remove_cvref

**Review by ferdonline — APPROVED** [review]

**Review by ohm314 — APPROVED** [review]
lgtm

File: include/highfive/bits/H5Utils.hpp (hunk 2 above)
Comment (ferdonline) [reply include/highfive/bits/H5Utils.hpp:42]: LGTM as well

Comment (matz-e) [reply include/highfive/bits/H5Utils.hpp:42]: Well, I hear complaints about
```
):
```
In Python code… so `>::type;` looks just as weird to me :)

**Pull request merged**